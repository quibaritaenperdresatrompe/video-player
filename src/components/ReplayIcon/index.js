import glamorous from 'glamorous'
import PropTypes from 'prop-types'
import React, { PureComponent } from 'react'

const StyledSvg = glamorous.svg(props => ({
  fill: props.color,
  ':hover': {
    fill: props.backgroundColor,
  },
}))

class ReplayIcon extends PureComponent {
  render() {
    return (
      <StyledSvg
        backgroundColor={this.props.backgroundColor}
        color={this.props.color}
        height={this.props.size}
        viewBox='0 0 24 24'
        width={this.props.size}
      >
        <path fill='none' d='M0 0h24v24H0z'/>
        <path d='M12 5V1L7 6l5 5V7c3.31 0 6 2.69 6 6s-2.69 6-6 6-6-2.69-6-6H4c0 4.42 3.58 8 8 8s8-3.58 8-8-3.58-8-8-8z'/>
      </StyledSvg>
    )
  }
}

ReplayIcon.propTypes = {
  backgroundColor: PropTypes.string,
  color: PropTypes.string,
  size: PropTypes.string,
}

ReplayIcon.defaultProps = {
  backgroundColor: '#d42929',
  color: '#fff',
  size: '1em',
}

export default ReplayIcon
