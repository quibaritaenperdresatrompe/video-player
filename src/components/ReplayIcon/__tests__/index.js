import ReplayIcon from '../'

import itRendersAllMutations from '../../../utils/jest-it-renders-all-mutations'

const mutations = [
  {
    name: 'without props',
    props: null,
  },
  {
    name: 'with all props',
    props: {
      backgroundColor: '#000',
      color: '#fff',
      size: '3em',
    },
  },
]

describe(ReplayIcon.name, () => {
  itRendersAllMutations(ReplayIcon, mutations)
})
