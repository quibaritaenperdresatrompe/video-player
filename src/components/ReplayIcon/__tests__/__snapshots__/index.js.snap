// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReplayIcon with all props it renders correctly 1`] = `
<svg
  className="css-1hgjwxk"
  color="#fff"
  height="3em"
  viewBox="0 0 24 24"
  width="3em"
>
  <path
    d="M0 0h24v24H0z"
    fill="none"
  />
  <path
    d="M12 5V1L7 6l5 5V7c3.31 0 6 2.69 6 6s-2.69 6-6 6-6-2.69-6-6H4c0 4.42 3.58 8 8 8s8-3.58 8-8-3.58-8-8-8z"
  />
</svg>
`;

exports[`ReplayIcon without props it renders correctly 1`] = `
<svg
  className="css-189tqkb"
  color="#fff"
  height="1em"
  viewBox="0 0 24 24"
  width="1em"
>
  <path
    d="M0 0h24v24H0z"
    fill="none"
  />
  <path
    d="M12 5V1L7 6l5 5V7c3.31 0 6 2.69 6 6s-2.69 6-6 6-6-2.69-6-6H4c0 4.42 3.58 8 8 8s8-3.58 8-8-3.58-8-8-8z"
  />
</svg>
`;
